net/adguardhome/Makefile

```diff
@@ -6,13 +6,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=adguardhome
-PKG_VERSION:=0.107.53
+PKG_VERSION:=0.107.54
 PKG_RELEASE:=1
 
-PKG_SOURCE_PROTO:=git
-PKG_SOURCE_VERSION:=v$(PKG_VERSION)
-PKG_SOURCE_URL:=https://github.com/AdguardTeam/AdGuardHome
-PKG_MIRROR_HASH:=d74702bc4f8b82bda64a0a937a98e73ee602c21b9361c0c683671212e03e9316
+PKG_SOURCE:=v$(PKG_VERSION).tar.gz
```

Comment:
No, please check how other packages have done for this.

Reply:
It feels so nasty when we have a tarball for this but you just cant use it because of [download.pl](https://github.com/openwrt/openwrt/blob/main/scripts/download.pl)
And it's even worse when the git script is doing a full clone from the repository, it's almost 3x larger than the tarball from tags

Reply:
Does the size matter on the computer, which you are using for the buildsystem to evenutally to cross-compile packages for your devices?  :-) I did not verify the size so far of your proposed solution, which says use upstream tag tarballs without the reason, why instead of the Git repo.

However, using Git sources is not preferred at least from my point of view. Its a big security risk, because if someone tinkers with the tarball, checksum is not verified anyhow. See: https://lists.openwrt.org/pipermail/openwrt-devel/2024-April/042521.html 

In your case, you might want to use codeload, though.  If we are talking about using codeload or your proposed solution, dont forget, that the tarball is created by GitHub itself, which means that there isnt checksum to verify (see checksums.txt file) in their release. Sometimes autogenerated tarballs are missing some things, which are required to have to build the package and the tarball from GitHub can be changed anyhow and then the checksum is not valid (I havent experienced this case, yet, but others said yes...)

So, I dont find anything wrong to keep using Git sources, i still dont know the reason, why you did it. 🤷 

We are not living in the ideal world.

Reply:
Please do not repeat the same mistakes with libxz issue

> And it's even worse when the git script is doing a full clone from the repository, it's almost 3x larger than the tarball from tags

also, why is the git script not doing shallow clones? does it pull the entire commit history?

Reply:
@vincejv May I ask what are you talking about, the xz backdoor incident? I can't understand why you are mentioning that here
And yes, in my case the script is doing a full clone.
```
Cloning into 'adguardhome-0.107.54'...
remote: Enumerating objects: 38166, done.
remote: Counting objects: 100% (76/76), done.
remote: Compressing objects: 100% (49/49), done.
Receiving objects:   1% (514/38166), 204.00 KiB | 15.00 KiB/s
```

Reply:
@Ra2-IFV exactly what you are trying to do, getting the code from the tarballs instead of the repository and the tarball contained tainted and malicious backdoor code, which was NOT present at all in the repository, no one was able to check as reviewing source code from tarballs is not part of the code review process.

it was assumed the tarball = reviewed source code, but it was not, the maintainer injected code and blobs before signing and uploading the tarballs

See: https://lists.openwrt.org/pipermail/openwrt-devel/2024-April/042521.html

Reply:
@BKPepe because the script is doing a full clone, and for some reason my git keeps failing if trying to clone a very large repository, so I want to use tarballs instead.
Please forgive me for being so dumb, I still can't understand this... Even though the tarball is generated by Github, it's from a git tag, and in most cases it shouldn't change
We also have `PKG_HASH` to check malicious tarballs in case someone tries to do something nasty

Reply:
> @Ra2-IFV exactly what you are trying to do, getting the code from the tarballs instead of the repository and the tarball contained tainted and malicious backdoor code, which was NOT present at all in the repository, no one was able to check as reviewing source code from tarballs is not part of the code review process.

so the attacker uploaded a tarball by himself, and it's different from the repository?

> it was assumed the tarball = reviewed source code, but it was not, the maintainer injected code and blobs before signing and uploading the tarballs
> 
> See: https://lists.openwrt.org/pipermail/openwrt-devel/2024-April/042521.html\

But from the wikipedia and other sources, Jia Tan gained trust from other developers and seems the malicious codes are already in the repository, he just signed off a new release.. 
> Jia Tan gained the position of co-maintainer of [XZ Utils](https://en.wikipedia.org/wiki/XZ_Utils) and was able to sign off on version 5.6.0, which introduced the backdoor,

Reply:
Well let's back to the topic. I believe @BKPepe was talking about the filename at first.
Your PR will create a unidentifiable tarball with  just some random numbers and it _could_ lead potential conflict issue.

So again, please check other packages for example.

Reply:
Uh sorry, that's for another pull request but almost the same.